src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -451,6 +451,50 @@ private function isSimpleInstance($id, Definition $definition)
     }
 
     /**
+     * Checks if the definition is a trivial instance.
+     *
+     * @param Definition $definition
+     *
+     * @return bool
+     */
+    private function isTrivialInstance(Definition $definition)
```

Comment:
OK. My reasoning was that in the future, if you add new optimizations, the code will complicate a lot (`if ($this->isTrivialInstance() || $this->isServiceLengthSmall() || $this->isDevEnvironment())` vs `if ($this->canBeInlined())` and then adding more optimizations to that method).